Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,183 @@
+using Content.Shared.MobState.EntitySystems;
+using Content.Shared.Popups;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+
+namespace Content.Server.Guardian
+{
+    public class GuardianSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<GuardianCreatorComponent, UseInHandEvent>(OnGuardianCreated);
+            SubscribeLocalEvent<GuardianCreatorComponent, ExaminedEvent>(OnActivatorExamined);
+            SubscribeLocalEvent<GuardianComponent, MoveEvent>(OnGuardianMove);
+            SubscribeLocalEvent<GuardianHostComponent, MoveEvent>(OnGuardianHostMove);
+            SubscribeLocalEvent<GuardianComponent, DamageChangedEvent>(OnGuardianDamaged);
+            SubscribeLocalEvent<GuardianHostComponent, MobStateChangedEvent>(OnHostStateChange);
+        }
+
+        /// <summary>
+        /// Triggers when the host recives damage which puts the host in either critical or killed state
+        /// </summary>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (EntityManager.GetEntity(component.Hostedguardian) != null)
+            {
+                if (args.State.IsCritical())
+                {
+                    var guard = EntityManager.GetEntity(component.Hostedguardian);
+                    _popupSystem.PopupCoordinates(Loc.GetString("guardian-host-critical-warn"), guard.Transform.Coordinates, Filter.Entities(guard.Uid));
+                    SoundSystem.Play(Filter.Local(), "/Audio/Effects/guardian_warn.ogg", uid);
+                }
+                else if (args.State.IsDead())
+                {
+                    var guard = EntityManager.GetEntity(component.Hostedguardian);
+                    SoundSystem.Play(Filter.Pvs(guard), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.20f));
+                    EntityManager.QueueDeleteEntity(guard);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
+            }
+        }
+
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            if (EntityManager.GetEntity(uid).TryGetComponent<DamageableComponent>(out DamageableComponent? guardiandamage))
+            {
+                EntityManager.GetEntity(component.Host).TryGetComponent<DamageableComponent>(out DamageableComponent? hostdamage);
+                if (hostdamage != null)
+                {
+                    if (args.DamageDelta != null)
+                    {
+                        EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamagePercent));
```

Comment:
As above, also cache the system, also what's wrong with TryChangeDamage.

Reply:
IIRC it didn't work due to damage share mechanics being fucky on it